// Copyright 2023 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";
package worker_api;

// Next field number: 11
message WorkerParamsProto {
  reserved 2, 6, 7;

  message InputStrings {
    repeated string inputs = 1;
  }

  // Input
  bytes code = 1;
  oneof input_type {
    InputStrings input_strings = 8;
    // Used when InvocationRequest.treat_input_as_byte_str is true.
    bytes input_bytes = 9;
  }
  map<string, string> metadata = 3;
  optional bytes wasm = 4;

  // Output
  optional string response = 5;

  // Error details in case the program errored out.
  optional bytes error_message = 10;

  // The output from V8's Heap and Sample-based CPU profiler
  optional bytes profiler_output = 11;

  // Map of events to the duration (in milliseconds) they take to execute.
  map<string, double> metrics = 12;
}
